### DISCUSSION TITLE
Refresh ui.table for every client at the same time?

### DISCUSSION THREAD

**Discussion by pythonormen** [zauberzeug/nicegui#1502] [Q&A] [answered]
### Question

Sorry in advance if I have asked this before but is there any way outside ui.refreshable (not an option as pagination etc is lost) to refresh table content for every client at the same time?
Like to be able to trigger the update for every client from outside the page decorator?

I have a 6s timer to refresh content, but would be very useful if changes would update immediately for all users if someone e.g. deletes a row.

Otherwise I guess ui.refreshable might be doable with adjustments similar to https://github.com/zauberzeug/nicegui/discussions/1029 but it sounds very advanced to maintain pagination etc.

Thanks!

```python
from nicegui import ui
import random


@ui.page('/')
async def main():
    async def table_update():
        table.rows[:] = [
            {'name': 'Alice', 'age': random.randint(0, 100)},
            {'name': 'Bob', 'age': random.randint(0, 100)},
            {'name': 'Carol'},
        ]
        table.update()


    columns = [
        {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
        {'name': 'age', 'label': 'Age', 'field': 'age', 'sortable': True},
    ]
    
    rows = [
            {'name': 'Alice', 'age': 1},
            {'name': 'Bob', 'age': 2},
            {'name': 'Carol'},
        ]
    
    table = ui.table(columns=columns, rows=rows, row_key='name')
    
    ui.timer(interval=3, callback=table_update)
    

ui.run(host='127.0.0.1')
```

**Comment by falkoschindler** [comment, marked as answer]
Good question, @pythonormen!

I came up with the following approach:

```py
from nicegui import Client, ui

tables: list[ui.table] = []
columns = [{'name': 'age', 'label': 'Age', 'field': 'age'}]
rows = [{'age': 0}]

@ui.page('/')
async def main(client: Client):
    table = ui.table(columns=columns, rows=rows)
    tables.append(table)
    await client.disconnected()
    tables.remove(table)

def update():
    rows[0]['age'] += 1
    for table in tables:
        table.update()

ui.timer(1, callback=update)

ui.run()
```

This keeps a shared state of rows and columns as well as a list of table elements. For every new client, a new table is added to this list. When the data changes, all tables are updated. We just need to make sure to remove a table once its client disconnects. This can be done by awaiting its disconnection. Registering a disconnect handler is also possible:
```py
client.on_disconnect(lambda: tables.remove(table))
```
